zingstats/changes.py

```diff
@@ -0,0 +1,287 @@
+#
+# (c) Copyright 2018 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import base64
+import json
+import logging
+import urllib
+from datetime import datetime
+
+import requests
+
+log = logging.getLogger(__name__)
+
+
+class Message(object):
+    def __init__(self, message_id, message_date, message_text):
+        self.message_id = message_id
+        self.message_dt = None
+        self.text = message_text
+
+
+class GerritMessage(Message):
+    def __init__(self, message_id, message_date, message_text):
+        super(GerritMessage, self).__init__(message_id, message_date,
+                                            message_text)
+        self.message_id = message_id
+        self._message_ts = message_date[:-3]
+        self.message_dt = datetime.strptime(self._message_ts,
+                                            GerritChange.GERRIT_FORMAT)
+
+
+class Revision(object):
+    def __init__(self, revision_id, revision, url, verify_https_requests):
+        self.revision_id = revision_id
+        # revision url is not directly usable, but a base for other queries
+        self.url = url
+        self.verify_https_requests = verify_https_requests
+        self._files = dict()
+        self._messages = list()
+        self.number = None
+        self.created_dt = None
+
+    def files(self):
+        for file_name in sorted(self._files):
+            yield file_name, self._files[file_name]['diff'],\
+                  self._files[file_name]['content']
+
+    def add_message(self, message):
+        self._messages.append(message)
+
+    def messages(self):
+        for message in sorted(self._messages):
+            yield message
+
+
+class GerritRevision(Revision):
+    def __init__(self, revision_id, revision, url=None,
+                 verify_https_requests=False):
```

Comment:
Might the verify capability be more neatly implemented by passing a requests.Session object?  Would allow for better extensibility too, I guess (e.g. if you needed to add authentication).

Reply:
Yes, will do that